mlx_lm/server.py

```diff
@@ -532,10 +532,18 @@ def handle_completion(
 
             if self.logprobs > 0:
                 sorted_indices = mx.argpartition(-logprobs, kth=self.logprobs - 1)
-                top_indices = sorted_indices[: self.logprobs]
-                top_logprobs = logprobs[top_indices]
-                top_token_info = zip(top_indices.tolist(), top_logprobs.tolist())
-                top_tokens.append(tuple(top_token_info))
+                top_indices = sorted_indices[
+                    : self.logprobs
+                ].tolist()  # Convert to Python list
+                top_logprobs = logprobs[top_indices].tolist()  # Convert to Python list
+                top_tokens.append(
+                    {
+                        int(idx): float(prob)
+                        for idx, prob in zip(
+                            top_indices, top_logprobs
+                        )  # Iterate over Python lists
+                    }
+                )
```

Comment:
Is this loop necessary? Just do:

```suggestion
                top_tokens.append(tuple(zip(top_indices, top_logprobs)))
```